src/adapter/src/coord.rs

```diff
@@ -459,10 +473,11 @@ pub struct PeekStageRealTimeRecency {
     timeline_context: TimelineContext,
     oracle_read_ts: Option<Timestamp>,
     in_immediate_multi_stmt_txn: bool,
-    optimizer: optimize::peek::Optimizer,
-    global_mir_plan: optimize::peek::GlobalMirPlan,
-    /// Context from where this peek initiated.
-    peek_ctx: PeekContext,
+    optimizer: Either<optimize::peek::Optimizer, optimize::copy_to::Optimizer>,
+    global_mir_plan: Either<optimize::peek::GlobalMirPlan, optimize::copy_to::GlobalMirPlan>,
```

Comment:
Just pulled the changes from https://github.com/MaterializeInc/materialize/pull/24903.

cc @aalexandrov @mjibson Instead of using `Either` would generic traits be better here ( I will take a stab at it then)? Because it's more tightly coupled than implied here, because `peek::Optimizer` can only be associated with `peek::GlobalMirPlan`.

Reply:
I went with `Either` because it is the least cognitive overhead. I'm not sure that introducing traits with associated types will make this easier to maintain - at some point we might want to change something for the `peek` branch and not for the `copy_to` branch in a place where the trait-based design dictates that both branches should be defined in terms of the same generic source code. That being said I'm not opposed to the approach if somebody else wants to take a stab.